src/gui/qgspointcloudquerybuilder.cpp

```diff
@@ -0,0 +1,317 @@
+  buttonBox->addButton( pbn, QDialogButtonBox::ActionRole );
+  pbn->setToolTip( tr( "Save query to QQF file" ) );
+  connect( pbn, &QAbstractButton::clicked, this, &QgsPointCloudQueryBuilder::saveQuery );
+
+  pbn = new QPushButton( tr( "&Load…" ) );
+  buttonBox->addButton( pbn, QDialogButtonBox::ActionRole );
+  pbn->setToolTip( tr( "Load query from QQF file" ) );
+  connect( pbn, &QAbstractButton::clicked, this, &QgsPointCloudQueryBuilder::loadQuery );
+
+  mOrigSubsetString = layer->subsetString();
+
+  lblDataUri->setText( tr( "Set provider filter on %1" ).arg( layer->name() ) );
+  mTxtSql->setText( mOrigSubsetString );
+}
+
+void QgsPointCloudQueryBuilder::showEvent( QShowEvent *event )
+{
+  mTxtSql->setFocus();
+  QDialog::showEvent( event );
+}
+
+void QgsPointCloudQueryBuilder::setupGuiViews()
+{
+  //Initialize the models
+  mModelAttributes = new QStandardItemModel();
+  mModelValues = new QStandardItemModel();
+
+  // Modes
+  lstAttributes->setViewMode( QListView::ListMode );
+  lstValues->setViewMode( QListView::ListMode );
+  lstAttributes->setSelectionBehavior( QAbstractItemView::SelectRows );
+  lstValues->setSelectionBehavior( QAbstractItemView::SelectRows );
+  lstAttributes->setEditTriggers( QAbstractItemView::NoEditTriggers );
+  lstValues->setEditTriggers( QAbstractItemView::NoEditTriggers );
+  // Performance tip since Qt 4.1
+  lstAttributes->setUniformItemSizes( true );
+  lstValues->setUniformItemSizes( true );
+  // Colored rows
+  lstAttributes->setAlternatingRowColors( true );
+  lstValues->setAlternatingRowColors( true );
+
+  lstAttributes->setModel( mModelAttributes );
+  lstValues->setModel( mModelValues );
+}
+
+void QgsPointCloudQueryBuilder::populateAttributes()
+{
+  const QgsFields &fields = mLayer->dataProvider()->index()->attributes().toFields();
+  mTxtSql->setFields( fields );
+  for ( int idx = 0; idx < fields.count(); ++idx )
+  {
+    QStandardItem *myItem = new QStandardItem( fields.at( idx ).displayNameWithAlias() );
+    mModelAttributes->insertRow( mModelAttributes->rowCount(), myItem );
+  }
+}
+
+void QgsPointCloudQueryBuilder::lstAttributes_currentChanged( const QModelIndex &current, const QModelIndex &previous )
+{
+  Q_UNUSED( previous )
+
+  mModelValues->clear();
+  const QString attribute = current.data().toString();
+  if ( attribute.compare( QLatin1String( "Classification" ), Qt::CaseInsensitive ) == 0 )
+  {
+    const QMap<int, QString> codes = QgsPointCloudDataProvider::translatedLasClassificationCodes();
+    for ( int i = 0; i <= 18; ++i )
+    {
+      QStandardItem *item = new QStandardItem( QString( "%1: %2" ).arg( i ).arg( codes.value( i ) ) );
+      item->setData( i, Qt::UserRole );
+      mModelValues->insertRow( mModelValues->rowCount(), item );
+    }
+  }
+  else
+  {
+    QVariant value = mLayer->dataProvider()->metadataStatistic( attribute, QgsStatisticalSummary::Min );
+    QString valueString = value.isNull() ? QLatin1String( "n/a" ) : value.toString();
```

Comment:
Do we need to translate "n/a"? (My gut feeling is yes, but I'm also not involved in the translation efforts). (ping @rduivenvoorde @DelazJ )

Reply:
Uh, yes, if that is seen by the user?
 "n/a" is written differently in Japanese :-)

Reply:
Fixed it, also added missing `tr`s to `minimum` `maximum` etc